src/main/groovy/com/bmuschko/gradle/docker/utils/RegistryAuthLocator.groovy

```diff
@@ -0,0 +1,293 @@
+package com.bmuschko.gradle.docker.utils
+
+import com.bmuschko.gradle.docker.DockerRegistryCredentials
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.dockerjava.api.model.AuthConfig
+import groovy.transform.CompileStatic
+import org.apache.commons.lang.SystemUtils
```

Comment:
This library might be pulled in by Docker Java, however, I'd like to avoid dependencies to other external libraries as much as possible. Can you maybe just inline the logic?